### ISSUE TITLE
{map,cache} sinks could react on null value

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast-jet#608]
I am trying to create a map pump. Something along these lines:
```java
        pipeline.drawFrom(mapJournal(sourceMapName, false))
                .map(e -> new AbstractMap.SimpleImmutableEntry(e.getKey(), e.getNewValue()))
                .drainTo(Sinks.map(targetMapName));
```

but I suspect this wont work where an entry is removed from the source map. Perhaps the sink could call `map.remove()` under the hood when it receives `null` as a value?

**Comment by cangencer** [comment]
You can add a filter for the event types you are interested in. I do think we can add some kind flags to register the event types you are interested in when you create the event journal though. Doing it via a predicate gets quite verbose. 

```java
SourceProcessors.<String, Tuple2<Integer, Long>, PriceUpdateEvent>streamMapP(
                        "prices",
                        e -> e.getType() == EntryEventType.ADDED || e.getType() == EntryEventType.UPDATED,
                        e -> new PriceUpdateEvent(e.getKey(), e.getNewValue().f0(), e.getNewValue().f1()),
                        true)```

**Comment by jerrinot** [comment]
but I am actually interested in the REMOVED events. as I want to removed them from the target map.

**Comment by viliam-durina** [comment]
This is now implemented as part of #621: you can use `mapWithMerging` to delete entries from the map.
